### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'regularRegex')

### ISSUE THREAD

**Issue by muellerj93** [mikezaschka/cds-dbm#314]
Hi @frankmeertens / @mikezaschka
as suggested in https://github.com/mikezaschka/cds-dbm/issues/309 I also tried using the latest versions. It works locally, but when I try to deploy to the BTP, it still gives me the following error with the callstack:

```
2023-03-10T15:04:11.39+0100 [APP/TASK/deploy_to_postgresql/0] OUT [cds-dbm] - starting delta database deployment of service db
   2023-03-10T15:04:32.23+0100 [APP/TASK/deploy_to_postgresql/0] OUT [cds-dbm] - delta successfully deployed to the database
   2023-03-10T15:04:32.59+0100 [APP/TASK/deploy_to_postgresql/0] OUT [cds-dbm] - loading data from /home/vcap/app/data/schwarz.sci.monitoring-MetricTypesList.csv
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR cds-dbm deploy [services]
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR Dynamically identifies changes in your cds data model and deploys them to the
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR database
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR Options:
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR --help           Show help                                       [boolean]
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR --version        Show version number                             [boolean]
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR -s, --service                                                [default: ["db"]]
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR -a, --auto-undeploy
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR -d, --dry
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR -l, --load-via
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR -c, --create-db
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR TypeError: Cannot read properties of undefined (reading 'regularRegex')
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at smartId (/home/vcap/deps/1/node_modules/@sap/cds-compiler/lib/sql-identifier.js:79:9)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at _smartId (/home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/common/utils/quotingStyles.js:9:23)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at PGReferenceBuilder.plain [as _quoteElement] (/home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/common/utils/quotingStyles.js:23:12)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at /home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/db/sql-builder/ReferenceBuilder.js:97:49
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at Array.map (<anonymous>)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at PGReferenceBuilder._parseReference (/home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/db/sql-builder/ReferenceBuilder.js:97:34)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at PGReferenceBuilder.build (/home/vcap/deps/1/node_modules/cds-pg/lib/pg/sql-builder/ReferenceBuilder.js:44:52)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at PGSelectBuilder._buildRefElement (/home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/db/sql-builder/SelectBuilder.js:276:68)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at PGSelectBuilder._buildElement (/home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/db/sql-builder/SelectBuilder.js:250:18)
   2023-03-10T15:04:32.64+0100 [APP/TASK/deploy_to_postgresql/0] ERR at /home/vcap/deps/1/node_modules/@sap/cds/libx/_runtime/db/sql-builder/SelectBuilder.js:315:56
```
After analyzing the callstack, the error should only occur when I pass a dialect that doesn't exist in the `/@sap/cds-compiler/lib/sql-identifier.js` `sqlDialects` Object.
However, as I'm passing `'plain'` it should work, especially, since the calling function `@sap/cds/libx/_runtime/common/utils/quotingStyles.js` makes sure to not pass an undefined `_dialect` variable.

I'm really confused as to why this error occurs only when deploying to BTP.

Below you can find my cds config, in which I don't see any error:

```json
...
"db": {
            "[production]": {
                "kind": "database",
                "dialect": "plain",
                "impl": "cds-pg",
                "model": ["srv"]
            },
            "[development]": {
                "kind": "database",
                "dialect": "plain",
                "impl": "cds-pg",
                "model": ["srv"],
                "credentials": {
                    "host": "localhost",
                    "port": 5432,
                    "database": "monitoring",
                    "user": "postgres",
                    "password": "postgres"
                }
            },
...
```

Unfortunately I have no idea on how to analyze what's happening during the deployment other than checking the logs / callstack.
Do you have an idea why this might happen?

Thanks in advance
Regards

**Comment by frankmeertens** [comment]
I seems to be related to the following where the `dialect` is not properly defined. Checking where this is coming from during deployment.

https://github.com/sapmentors/cds-pg/issues/285

**Comment by frankmeertens** [comment]
@mikezaschka The `package.json` in the `build/postgres-cf/template` folder will work when the dialect is added. Not sure yet whether this is the fix that will be applied in `cds-pg`, but seems to resolve the issue. 

Example:
```JSON
  "cds": {
    "requires": {
      "db": {
        "kind": "database",
        "dialect": "plain"
      },
      "database": {
        "dialect": "plain",
        "impl": "cds-pg",
        "model": "csn.json"
      }
    },
```

**Comment by frankmeertens** [comment]
Was able to test deployment with the delta load option and was successful. Created a pull request with the code changes in case that helps?

**Comment by mikezaschka** [comment]
Thanks!
I merged https://github.com/mikezaschka/cds-dbm/pull/315 and published cds-dbm@0.1.4.
